site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1168,13 +1168,16 @@ public function getActiveUserIds(bool $instructor, bool $fullAccess, bool $limit
             $extra_join = ' INNER JOIN users ON courses_users.user_id = users.user_id ';
             $extra_where = ' OR users.user_access_level <= 2';
         }
+        $terms = ($term === '' ? 'courses.term' : $term);
+        $courses = ($course === '' ? 'courses.course' : $course);
         $result_rows = [];
         $this->submitty_db->query(
             "SELECT DISTINCT courses_users.user_id as user_id
                 FROM courses_users INNER JOIN courses
-                ON courses_users.term = courses.term AND courses_users.course = courses.course
+                ON courses_users.term = ? AND courses_users.course = ? 
```

Comment:
This doesn't seem right to me.  Right now, the prepared statement means that this will perform the comparison `courses_users.term = 'courses.term'` if there is no term, when you presumably mean `courses_users.term = courses.term` as originally written.  You could instead move the logic entirely into the SQL query, or conditionally insert `courses.term` or `?`, depending on whether the term was empty.